mono/mini/mini-runtime.c

```diff
@@ -4548,41 +4548,97 @@ typedef struct _InstalledRuntimeCode InstalledRuntimeCode;
 
 
 static InstalledRuntimeCode*
-ves_icall_mjit_install_compilation_result (int compilation_result, NativeCodeHandle *native_code)
+ves_icall_mjit_install_compilation_result (int compilation_result, NativeCodeHandle native_code)
 {
+	ERROR_DECL (error);
+	gpointer params [0] = {0};
+
 	if (compilation_result != 0) {
 		g_printerr ("mjit_install: failed with %d\n", compilation_result);
 		return NULL;
 	}
+	static MonoClass *MethodInfo_klass = NULL;
+	static MonoMethod *MethodInfo_get_RuntimeMethodHandle_method = NULL;
+
+	g_assertf (mono_defaults.compiler, "Mono.Compiler must be initialized by runtime init");
+
+	if (!MethodInfo_klass) {
+		MethodInfo_klass = mono_class_from_name_checked (mono_defaults.compiler, "Mono.Compiler", "MethodInfo", error);
+		return_val_if_nok (error, NULL);
+
+		g_assert (MethodInfo_klass);
+	}
+
+	if (!MethodInfo_get_RuntimeMethodHandle_method) {
+		MethodInfo_get_RuntimeMethodHandle_method = mono_class_get_method_from_name_checked (MethodInfo_klass, "get_RuntimeMethodHandle", 0, 0, error);
```

Comment:
we could add a real backing field and a struct on the C side.

Reply:
so I tried that first and somehow I failed miserably.  I didn't investigate any further as it worked with this approach just fine, but yes, it's a bit expensive.